src/pages/build-apps/guides/authentication.md

```diff
@@ -0,0 +1,263 @@
+
+An app and authenticator, such as [the Stacks Wallet](https://blockstack.org/wallet), communicate during the authentication flow by passing back and forth two tokens. The requesting app sends the authenticator an `authRequest` token. Once a user approves authentication, the authenticator responds to the app with an `authResponse` token.
+
+These tokens are are based on [a JSON Web Token (JWT) standard](https://tools.ietf.org/html/rfc7519) with additional support for the `secp256k1` curve used by Bitcoin and many other cryptocurrencies. They are passed via URL query strings.
+
+See the [`authRequest`](#authrequest-payload-schema) and [`authResponse`](#authresponse-payload-schema) payload schemas below for more details about what data they contain.
+
+When a user chooses to authenticate an app, it sends the `authRequest` token to the authenticator via a URL query string with an equally named parameter:
+
+`https://wallet.hiro.so/...?authRequest=j902120cn829n1jnvoa...`
+
+When the authenticator receives the request, it generates an `authResponse` token for the app using an _ephemeral transit key_ . The ephemeral transit key is just used for the particular instance of the app, in this case, to sign the `authRequest`.
+
+The app stores the ephemeral transit key during request generation. The public portion of the transit key is passed in the `authRequest` token. The authenticator uses the public portion of the key to encrypt an _app private key_ which is returned via the `authResponse`.
+
+The authenticator generates the app private key from the user's _identity address private key_ and the app's domain. The app private key serves three functions:
+
+1. It is used to create credentials that give the app access to a storage bucket in the user's Gaia hub
+2. It is used in the end-to-end encryption of files stored for the app in the user's Gaia storage.
+3. It serves as a cryptographic secret that apps can use to perform other cryptographic functions.
+
+Finally, the app private key is deterministic, meaning that the same private key will always be generated for a given Stacks address and domain.
+
+The first two of these functions are particularly relevant to [data storage with Stacks.js](/build-apps/guides/data-storage).
+
+[Learn more about keypairs](#key-pairs) used by authentication.
+
+## Install dependency
+
+The following dependency must be installed:
+
+```
+npm install @stacks/connect
+```
+
+## Initiate userSession object
+
+Apps keep track of user authentication state with the `userSession` object, initiated with the `UserSession` and `AppConfig` classes:
+
+```js
+import { AppConfig, UserSession } from '@stacks/connect';
+
+const appConfig = new AppConfig(['store_write', 'publish_data']);
+const userSession = new UserSession({ appConfig });
+```
+
+The main thing to decide here is what permission scopes your app needs from the user during authentication.
+
+Apps may request any of the following scopes:
```

Comment:
technically these are the available scopes:
```tsx
export declare const enum AuthScope {
    store_write = "store_write",
    publish_data = "publish_data",
    email = "email"
}
```

cc @hstove

Reply:
I believe we removed `email` support from in the Stacks Wallet at least. Does it remain in the Stacks.js/connect package in a meaningful way?

Reply:
No - you can use it, but it won't do anything.